nMessenger/Source/Messenger/Components/NMessenger.swift

```diff
@@ -422,9 +422,11 @@ public class NMessenger: UIView {
      after the block
      - parameter competion: called once the semaphore has expired
      */
-    private func waitForMessageLock(completion: ()->Void) {
-        dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0)) {
-            dispatch_semaphore_wait(self.state.messageLock, DISPATCH_TIME_FOREVER)
+    fileprivate func waitForMessageLock(_ completion: @escaping ()->Void) {
+        //DispatchQueue.global(priority: DispatchQueue.GlobalQueuePriority.default).async {
+            
+        DispatchQueue.global(qos: DispatchQoS.QoSClass.default).async {
```

Comment:
This can also be `DispatchQueue.global().async {}`. (qos' default value is ´DispatchQoS.QoSClass.default`) I'd prefer this since it is a bit cleaner.